haigha/reader.py

```diff
@@ -231,8 +231,24 @@ def read_timestamp(self):
 
         Will raise BufferUnderflow if there's not enough bytes in the buffer.
         Will raise struct.error if the data is malformed
+
+        To support different RabbitMQ constructs, the timestamp is parsed in different basis from seconds to microseconds.
         """
-        return datetime.utcfromtimestamp(self.read_longlong())
+        ts = self.read_longlong()
+        try:
+            # Try to parse the timestamp in seconds
+            return datetime.utcfromtimestamp(ts)
+        except:
```

Comment:
Such wide-open, general `except:` statements are frowned upon in python development. They hide programming errors, such as invalid parameters, etc.